lib/recent.go

```diff
@@ -0,0 +1,92 @@
+package lib
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type RecentFiles struct {
+	Terraform []string `json:"terraform"`
+	Tofu      []string `json:"tofu"`
+}
+
+func addRecent(requestedVersion string, installPath string, dist string) {
+	if !validVersionFormat(requestedVersion) {
+		logger.Errorf("The version %s is not a valid version string and won't be stored", requestedVersion)
+		return
+	}
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	if CheckFileExist(recentFilePath) {
+		unmarshal(recentFilePath, &recentFileData)
+	}
+	var sliceToCheck []string
+	if dist == distTerraform {
+		sliceToCheck = recentFileData.Terraform
+	} else if dist == distTofu {
+		sliceToCheck = recentFileData.Tofu
+	}
+	for _, v := range sliceToCheck {
+		if v == requestedVersion {
+			// entry already exists. Nothing to do
+			return
+		}
+	}
+	if dist == distTerraform {
+		recentFileData.Terraform = append(recentFileData.Terraform, requestedVersion)
+	} else if dist == distTofu {
+		recentFileData.Tofu = append(recentFileData.Tofu, requestedVersion)
+	}
+	saveFile(recentFileData, recentFilePath)
+}
+
+func getRecentVersions(installPath string, dist string) ([]string, error) {
+	installLocation := GetInstallLocation(installPath)
+	recentFilePath := filepath.Join(installLocation, recentFile)
+	var recentFileData RecentFiles
+	unmarshal(recentFilePath, &recentFileData)
+	if dist == distTerraform {
+		return recentFileData.Terraform, nil
+	} else if dist == distTofu {
```

Comment:
For visibility in this PR (from https://github.com/warrensbox/terraform-switcher/pull/436):
I'll just need to work on integrating these changes into https://github.com/warrensbox/terraform-switcher/pull/435, as this PR was based on the branch for that :)

I guess, adding the changes for OpenTofu might be a little.. odd, if https://github.com/warrensbox/terraform-switcher/pull/437 is merged before that.

If you did want to merge this PR prior to https://github.com/warrensbox/terraform-switcher/pull/435 , then I think it makes sense (though the references to OpenTofu are possibly strange), but if it does depend on it and/or makes more sense to ship together, is it worth either branching from https://github.com/warrensbox/terraform-switcher/pull/435 and making the changes or even branching from this PR to make the suggested changes?